### PR TITLE
[12.0][account] try to link moves with their reversals

### PR DESCRIPTION
**Pull request by JordiBForgeFlow**
Even if you have account_reversal installed, it may well be that there are old reversals that were created without this module. This PR will try to link all moves with their corresponding reversals, excluding those that were created using module 'account_reversal'.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by MiquelRForgeFlow — CHANGES_REQUESTED** [review]
This fix has no sense. There is no `for`. Only there is an `if` and an `else`.

**Comment by JordiBForgeFlow** [comment]
@mreficent the if checks wether you habe the account_reversal module installed. But if you have it installed, and you have prior reversals made without this module, then the old reversals will not be fixed.

This PR fixes that, and I checked that it solves the issue it describes.

File: addons/account/migrations/12.0.1.1/post-migration.py [hunk 1]
```diff
@@ -167,14 +168,17 @@ def fill_account_move_reverse_entry_id(env):
...
     else:
         installed_langs = env['res.lang'].search([])  # search only active
         for lang in installed_langs:
             reversal_text = env['ir.translation']._get_source(
                 name=False, types='code', lang=lang.code,
                 source='reversal of: ')
             reversal_moves = all_moves.filtered(
-                lambda m: m.ref and m.ref.startswith(reversal_text))
+                lambda m: m.ref and m.ref.startswith(reversal_text)
+                and m not in moves_linked_to_reversals)
```
Comment (MiquelRForgeFlow) [inline addons/account/migrations/12.0.1.1/post-migration.py]: here `moves_linked_to_reversals` will be always empty, so it's irrelevant adding `m not in moves_linked_to_reversals`

File: addons/account/migrations/12.0.1.1/post-migration.py [hunk 2]
```diff
@@ -167,14 +168,17 @@ def fill_account_move_reverse_entry_id(env):
             'domain': [('auto_reverse', '=', True),
                        ('reverse_date', '=', '2999-12-31')],
         })
+        moves_linked_to_reversals += all_moves.filtered(
```
Comment (mreficent) [inline addons/account/migrations/12.0.1.1/post-migration.py]: and here you are filling `moves_linked_to_reversals` but without any use

**Review by MiquelRForgeFlow — CHANGES_REQUESTED** [review]

**Comment by MiquelRForgeFlow** [comment]
Fixed :+1:

**Review by MiquelRForgeFlow — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Added improvement for better performance.

File: addons/account/migrations/12.0.1.1/post-migration.py [hunk 3]
```diff
@@ -156,31 +156,31 @@ def fill_account_journal_alias_id(env):
...
 def fill_account_move_reverse_entry_id(env):
-    all_moves = env['account.move'].search([])
-    moves_linked_to_reversals = env['account.move']
+    AccountMove = env['account.move']
     if openupgrade.table_exists(env.cr, 'account_move_reverse'):
-        reversal_moves = all_moves.filtered(
-            lambda m: m.auto_reverse and not m.reverse_entry_id)
+        reversal_moves = AccountMove.search([
+            ('auto_reverse', '=', True),
+            ('reverse_entry_id', '!=', False),
```
Comment (MiquelRForgeFlow) [inline addons/account/migrations/12.0.1.1/post-migration.py]: 'reverse_entry_id', **'='**, False

**Review by MiquelRForgeFlow — CHANGES_REQUESTED** [review]

**Review by pedrobaeza — APPROVED** [review]

**Comment by MiquelRForgeFlow** [comment]
I just fixed flake8

**Review by MiquelRForgeFlow — APPROVED** [review]

**Pull request merged**